src/download.c

```diff
@@ -500,6 +500,12 @@ static int poll_fewer_than(size_t xfer_queue_high, size_t xfer_queue_low)
 			}
 		}
 
+		// Do more work before processing the queue
+		curlm_ret = curl_multi_perform(mcurl, &running);
```

Comment:
This looks very similar to how the initial version was coded, with multiple curl_multi_perform()s happening. Looking thru the docs, would it make sense to check for CURLM_CALL_MULTI_PERFORM in these calls? There's also nothing wrong with just calling it multiple times "just in case" per the docs... not sure which is a better path.